torch/csrc/jit/codegen/cuda/lower_unroll.cpp

```diff
@@ -110,13 +110,25 @@ kir::Expr* findVectorizedSet(
 kir::Val* getVectorizeExtent(
     kir::TensorView* tv,
     const std::vector<kir::Val*>& indices) {
-  const auto gpu_lower = GpuLower::current();
-  kir::IrBuilder ir_builder(gpu_lower->kernel());
   auto domain = tv->domain()->hasRFactor() ? tv->domain()->rfactorDomain()
                                            : tv->domain()->rootDomain();
 
   TORCH_INTERNAL_ASSERT(domain.size() == indices.size());
 
+  bool is_contiguous = true;
+  for (auto status : tv->domain()->contiguity()) {
+    is_contiguous &= status;
+  }
+
+  // If the tensorview is not contiguous, return inner-most root domain extent
+  if (!is_contiguous) {
+    return domain.back()->extent();
```

Comment:
Why is this necessary? Do you have a test case that fails without this?

Reply:
We cannot merge non-contiguous root domains. Otherwise, we can get incorrect results.

Non-Contiguous Tensor:
```
[     row 0 - shift 0  ]       [     row 1 - shift 1  ]
[pre | vectorize | post] [***] [pre | vectorize | post]
                     ^-----------^ overlapping rows
```

If we merge non-contiguous root domains, we can index into the post and pre section of two different rows.
This would require two shift calculations, which isn't currently supported.

It is related to `FusionVectorizeMisalignedPointwiseMergeSymbolicPass_CUDA` and `FusionVectorizeMisalignedPointwiseMergeSymbolicFail_CUDA`.

Reply:
It only extends the vectorized domain when the index into a root axis is zero. IIUC, that happens iff the root axis is contiguous. I may be missing something, so I'm curious when it could fail without the added check. I commented out the `is_contiguous` part and didn't see any failure.

Reply:
Why is it necessary for there to be a failure to add this to the function?
I didn't see a reason to finish the function in this scenario.

Reply:
Sorry, I don't understand what you mean. Can you please rephrase your comment?

Reply:
See #826